File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
//         given to strings, replace all of the spaces to condense the words
        word1.replaceAll(" ", "");
        word2.replaceAll(" ", "");
//         get the max length so that way we can iterate to the very end
        int max = Math.max(word1.length(), word2.length());
        String merge = "";
//         iterate through, check the conditions, and concat to the string
        for(int i = 0; i < max; i++) {
            if (i < word1.length()) {
                merge += word1.charAt(i);
            }
            
            if (i < word2.length()) {
                merge += word2.charAt(i);
            }
        }
        return merge;
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    // word1.replace("/\s+/g", "").trim();
    // word2.replace("/\s+/g", "").trim();
    var maxLength = Math.max(word1.length, word2.length);
    var s = "";
    
    for (i = 0; i < maxLength; i++) {
        if (i < word1.length) {
           s += word1[i];
        }
        if (i < word2.length) {
            s += word2[i];
        }
    }
    
    return s;
};